app/samples/phase_detection/src/flight.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+#include <array>
+#include <cstdint>
+#include <f_core/flight/CPhaseController.h>
+
+/**
+ * Flight Events for a fictional flight
+ */
+enum Events : uint8_t { PadReady, Boost, Coast, Noseover, MainChute, GroundHit, CamerasOff, NumEvents };
+constexpr std::array<const char *, Events::NumEvents> eventNames = {
+    "PadReady", "Boost", "Coast", "Noseover", "MainChute", "GroundHit", "CamerasOff",
+};
+
+/**
+ * Sources of flight events
+ */
+enum Sources : uint8_t {
+    IMU1,
+    Barom1,
+    NoseoverLockout,
+    Boost2CoastTimer,
+    Noseover2MainTimer,
+    FullFlightTimer,
+    ExtraCameraTimer,
+    NumSources
+};
+inline constexpr std::array<const char *, Sources::NumSources> sourceNames = {
+    "IMU 1 (LSM6DSL)",     "Barom 1 (BME280)",  "Noseover Lockout",  "Boost2Coast Timer",
+    "Noseover2Main Timer", "Full Flight Timer", "ExtraCamera Timer",
+};
+
+inline constexpr std::size_t num_timer_events = 5;
```

Comment:
It would be nice to just infer this but wanted to create the alias Controller before having to write Controller::TimerEvent below rather than full template so here we are declaring it ahead of time.

Such is life